index.js

```diff
@@ -61,60 +64,93 @@ module.exports = app => {
+          started_at: timeStart,
+          conclusion: 'action_required',
+          completed_at: new Date(),
+          output: {
+            title: 'DCO',
+            summary
+          },
+          actions: [{
+            label: 'Set DCO to pass',
+            description: 'would set status to passing',
+            identifier: `override`
+          }]
+        }))
+          .catch(function checkFails (error) {
+            if (error.code === 403) {
+              console.log('resource not accessible, creating status instead')
+              // create status
+              const description = dcoFailed[(dcoFailed.length - 1)].message.substring(0, 140)
+              const params = {
+                sha: pr.head.sha,
+                context: 'DCO',
+                state: 'failure',
+                description,
+                target_url: 'https://github.com/probot/dco#how-it-works'
+              }
+              return context.github.repos.createStatus(context.repo(params))
             }
-            return context.github.repos.createStatus(context.repo(params))
+          })
+      } else {
+        context.github.checks.create(context.repo({
+          name: 'DCO',
+          head_branch: pr.head.ref,
+          head_sha: pr.head.sha,
+          status: 'completed',
+          started_at: timeStart,
+          conclusion: 'action_required',
+          completed_at: new Date(),
+          output: {
+            title: 'DCO',
+            summary
           }
-        })
+        }))
+          .catch(function checkFails (error) {
+            if (error.code === 403) {
+              console.log('resource not accessible, creating status instead')
+              // create status
+              const description = dcoFailed[(dcoFailed.length - 1)].message.substring(0, 140)
+              const params = {
+                sha: pr.head.sha,
+                context: 'DCO',
+                state: 'failure',
+                description,
+                target_url: 'https://github.com/probot/dco#how-it-works'
+              }
+              return context.github.repos.createStatus(context.repo(params))
+            }
+          })
+      }
     }
   }
 
-  // This option is only presented to users with Write Access to the repo
+  // This option is only presented to users with Write Access to the repo and config.enable_pass set to true
   app.on('check_run.requested_action', setStatusPass)
   async function setStatusPass (context) {
     const timeStart = new Date()
 
-    return context.github.checks.create(context.repo({
-      name: 'DCO',
-      head_branch: context.payload.check_run.check_suite.head_branch,
-      head_sha: context.payload.check_run.head_sha,
-      status: 'completed',
-      started_at: timeStart,
-      conclusion: 'success',
-      completed_at: new Date(),
-      output: {
-        title: 'DCO',
-        summary: 'Commit sign-off was manually approved.'
-      }
-    }))
+    if (enablePass) {
```

Comment:
`enablePass` was a variable you set in the context of the function `check`. `check` happens when commits are pushed and is based on webhook events. No state is maintained from there when the `check_run.requested_action` event is called, triggering this `setStatusPass` function.

You'll need to add something like https://github.com/probot/dco/pull/120/files#diff-168726dbe96b3ce427e7fedce31bb0bcR11 and https://github.com/probot/dco/pull/120/files#diff-168726dbe96b3ce427e7fedce31bb0bcR18 in this function.

Reply:
Good catch! Fixed now :-).

Reply:
Fixed the build error - please review

Reply:
@hiimbex - Can you review?

Reply:
@hiimbex can you review?

Reply:
Adding @brianwarner who also offered to review.